### PR TITLE
feat: lsc cli template

### PR DESCRIPTION
**Pull request by JamilOmar**
- upgrade of cli template with ts and tests

### CODE REVIEW

File: lib/cli/loader.ts [hunk 1]
```diff
@@ -216,14 +216,15 @@ export class CliLoader {
...
         if (!manifest) {
             return;
         }
 
-        let packageName = getPackageName(manifest),
-            commands = this.getCommands(directory, this.options.pattern);
-
+        const packageName = getPackageName(manifest);
+        const lscSettings = getPackageLSCSettings(manifest);
+        const pattern = (lscSettings &&  lscSettings.cliDir) ? `${lscSettings.cliDir}/*.js` : this.options.pattern;
```
Comment (artemnih) [inline lib/cli/loader.ts]: parenthesis are not required here

File: lib/cli/utils.ts [hunk 2]
```diff
@@ -7,17 +7,25 @@ const assert = require('assert');
 const glob = require('glob');
 const resolve = require('resolve-pkg');
 
+
+interface ILSCSettings {
```
Comment (artemnih) [inline lib/cli/utils.ts]: - interface names should **not** start with `I` letter
- abbreviations in names should be treated as words `ABCSettings` -> `AbcSettings`

File: lib/cli/utils.ts [hunk 3]
```diff
@@ -7,17 +7,25 @@ const assert = require('assert');
...
+    cliDir?: string
+    packageDependencies?: any|string[]
```
Comment (artemnih) [inline lib/cli/utils.ts]: is it possible to avoid `any`

File: lib/cli/utils.ts [hunk 4]
```diff
@@ -7,17 +7,25 @@ const assert = require('assert');
...
+}
+
 /**
  * @param manifest - A parsed LabShare package package.json file
  * @returns {Array} A list of LabShare package dependencies or an empty array
  * @private
  */
-function getPackageDependencies(manifest): string[] {
+function getPackageDependencies(manifest): any|string[] {
```
Comment (artemnih) [inline lib/cli/utils.ts]: is it possible to avoid `any`

File: lib/cli/utils.ts [hunk 5]
```diff
@@ -7,17 +7,25 @@ const assert = require('assert');
...
     return (_.isObject(manifest) && _.isObject(manifest.packageDependencies))
         ? (_.isArray(manifest.packageDependencies) ? manifest.packageDependencies : Object.keys(manifest.packageDependencies))
         : [];
 }
 
+ "<%= appNameSlug %>"
```
Comment (artemnih) [inline lib/cli/utils.ts]: is this out of place?

File: lib/cli/utils.ts [hunk 6]
```diff
@@ -28,6 +36,19 @@ export function getPackageName(manifest): string {
     }
     return (manifest.namespace || manifest.name).toLowerCase();
 }
+/**
+ * @description Retrieves the LabShare package's lsc settings
+ * @param {Object} manifest - A package.json parsed into a JS object
+ */
+export function getPackageLSCSettings(manifest): ILSCSettings {
```
Comment (artemnih) [inline lib/cli/utils.ts]: input type missing.
rename to follow camelCase https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)?redirectedfrom=MSDN

File: lib/commands/app.ts [hunk 7]
```diff
@@ -56,27 +56,12 @@ export const create = function () {
...
-          });
-          return;
-        }
+
         gulp.src([
             `${__dirname}/../../templates/common/**`,
             `${__dirname}/../../templates/${answers.projectType}-package/**`
         ])
-            .pipe(template(answers))
+            .pipe(template(answers , {'interpolate' : /<%=([\s\S]+?)%>/g}))
```
Comment (artemnih) [inline lib/commands/app.ts]: can you add a comment to understand the regex

File: lib/cli/loader.ts (hunk 1 above)
Comment (JamilOmar) [reply lib/cli/loader.ts]: makes the code more readable , it will stay like this

**Pull request merged**